backend/app/settle_usage.go

```diff
@@ -0,0 +1,191 @@
+package app
+
+import (
+	"kubecloud/internal"
+	"kubecloud/internal/logger"
+	"math"
+	"strconv"
+	"time"
+)
+
+type discount string
+
+type DiscountPackage struct {
+	DurationInMonth float64
+	Discount        int
+}
+
+// DeductBalanceBasedOnUsage deducts the user balance based on the usage
+// This function is called every 24 hours
+func (h *Handler) DeductBalanceBasedOnUsage() {
+	usageDeductionTicker := time.NewTicker(24 * time.Hour)
+	defer usageDeductionTicker.Stop()
+
+	for range usageDeductionTicker.C {
+		users, err := h.db.ListAllUsers()
+		if err != nil {
+			logger.GetLogger().Error().Err(err).Msg("Failed to list users")
+			continue
+		}
+
+		for _, user := range users {
+			usageInUSDMillicent, err := h.getUserDailyUsageInUSD(user.ID)
+			if err != nil {
+				logger.GetLogger().Error().Err(err).Msgf("Failed to get usage for user %d", user.ID)
+				continue
+			}
+
+			if err := h.db.DeductUserBalance(&user, usageInUSDMillicent); err != nil {
+				logger.GetLogger().Error().Err(err).Msgf("Failed to deduct balance for user %d", user.ID)
+			}
+		}
+	}
+}
+
+func (h *Handler) getUserDailyUsageInUSD(userID int) (uint64, error) {
+	now := time.Now()
+	// Define the end of the day (next day at 00:00)
+	endOfDay := time.Date(now.Year(), now.Month(), now.Day()+1, 0, 0, 0, 0, time.Local)
+
+	// Get the last calculation time for this user from the database, or use a default if not available
+	lastCalcTime, err := h.db.GetUserLastCalcTime(userID)
+	if err != nil {
+		return 0, err
+	}
+
+	// If this is the first time or no record exists, use the start of the day as default
+	if lastCalcTime.IsZero() {
+		lastCalcTime = time.Date(now.Year(), now.Month(), now.Day(), 0, 0, 0, 0, time.Local)
```

Comment:
Can we create the lastCalcTime field during user creation, as the service might fail or crash, and we will skip the user's payment for that day?